src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -26,33 +27,50 @@ object Scope {
     def getChildrenList(ast: Ast): Seq[Ast] = Option(ast.getAttribute("body")) collect {
       case list: AstList => list.asScala.toVector
     } getOrElse Seq.empty collect {
-      case ast: Ast => ast
+      case node: Ast
+        if node.getName == AstNodeName.Call ||
+          node.getName == AstNodeName.Scatter =>
+        node
+    }
```

Comment:
The only inconvenient I see is that it removes support for nested workflows but we do not support that anyway.
Also not directly related to this but the original idea of `generateTree` was to keep it as generic as possible so it could generate a Scope and its children for any Scope without any differentiation.
And `generateWorkflow` was merely a cast of `generateTree` to a workflow.